packages/runtime/src/runtime.ts

```diff
@@ -1036,4 +1038,70 @@ export class Runtime {
 			throw new RuntimeError(RUNTIME_ERRORS.GENERAL.INVALID_SECRET_KEY);
 		}
 	}
+
+	/**
+	 * Verifies multi-signature and throws an error if signatures are not valid
+	 * @param signedTransaction signedTransaction object
+	 */
+	validateMultisignature(signedTransaction: algosdk.SignedTransaction) {
+		this.verifyMultisig(signedTransaction);
+		if (!this.verifyMultisig(signedTransaction)) {
+			throw new RuntimeError(RUNTIME_ERRORS.GENERAL.INVALID_MULTISIG);
+		}
+	}
+
+	/**
+	 * Verify multi-signature
+	 * @param signedTxn signedTransaction object
+	 */
+	/* eslint-disable sonarjs/cognitive-complexity */
+	verifyMultisig(signedTxn: SignedTransaction): boolean {
+		if (signedTxn.msig === undefined) {
+			return false;
+		} else {
+			const version = signedTxn.msig.v;
+			const threshold = signedTxn.msig.thr;
+			const subsigs = signedTxn.msig.subsig;
+			const addrs = subsigs.map((subsig) => algosdk.encodeAddress(subsig.pk));
+			if (signedTxn.msig.subsig.length < threshold) {
+				return false;
+			}
+			let multiSigaddr;
+			try {
+				const mparams = {
+					version: version,
+					threshold: threshold,
+					addrs: addrs,
+				};
+				multiSigaddr = algosdk.multisigAddress(mparams);
+			} catch (e) {
+				return false;
+			}
+			const fromAccountAddr = webTx.getTxFromAddress(signedTxn.txn);
+			const fromAccount = this.getAccount(fromAccountAddr);
+			const accountSpendAddr = fromAccount.getSpendAddress();
+			if (accountSpendAddr !== multiSigaddr) {
+				return false;
+			}
+			let counter = 0;
+			for (const subsig of subsigs) {
+				if (!compareArray(subsig.s, new Uint8Array(0))) {
+					counter += 1;
+				}
+			}
+			if (counter < threshold) {
+				return false;
+			}
+			let verifiedCounter = 0;
+			for (const subsig of subsigs) {
+				if (
+					subsig.s &&
+					nacl.sign.detached.verify(signedTxn.txn.bytesToSign(), subsig.s, subsig.pk)
```

Comment:
This is a little bit new to me. How do you know that we can use `nacl` to verify signature?
Is there any source or docs that explain us to do that?

Reply:
this is what algosdk uses. Thanks why I am using it here. here are the docs http://nacl.cr.yp.to/